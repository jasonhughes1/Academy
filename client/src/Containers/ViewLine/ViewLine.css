.view-line-container {
  display: flex;
  flex-direction: column;
  background-color: #3C4040;
}

.place-holder {
  height: 100px;
}

.view-salary-title {
  margin-bottom: 50px;
  margin-top: 20px;
}
