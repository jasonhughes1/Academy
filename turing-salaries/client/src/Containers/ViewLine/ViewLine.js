import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import MyStockChart from '../highcharts/highcharts';
import PieChart from '../PieChart/PieChart';
import LineGraph from '../LineGraph/LineGraph';
import Pie from '../Pie/Pie';

class ViewLine extends Component {

  render() {
    return (
      <div className='view-salary-container'>
        <div className='link-container-view'>
        <NavLink className='nav' exact to='/'>Home</NavLink>
      </div>
        <h1 className='view-salary-title'>Salary and Job Comparisons</h1>
          <LineGraph alumData={this.props.alumni}/>
      </div>
    )
  }
}

export const mapStateToProps = (store) => {
  return {
    alumni: store.alumData
  };
};

export default connect(mapStateToProps, null)(ViewLine);
