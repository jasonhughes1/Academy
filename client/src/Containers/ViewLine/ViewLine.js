import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import LineGraph15 from '../LineGraphs/LineGraph15';
import LineGraph16 from '../LineGraphs/LineGraph16';
import LineGraphYearly from '../LineGraphs/LineGraphYearly'
import './ViewLine.css';


class ViewLine extends Component {

  render() {
    return (
      <div className='view-line-container'>
        <div className='link-container-view'>
        <NavLink className='nav' exact to='/'>Home</NavLink>
      </div>
        <h2 className='view-salary-title'>View salary information for cohorts from 2015 and 2016</h2>
          <LineGraph15 alumData={this.props.alumni}/>
          <div class="place-holder"></div>
          <LineGraph16 alumData={this.props.alumni}/>
          <LineGraphYearly alumData={this.props.alumni} />
          <div class="place-holder"></div>
      </div>
    )
  }
}

export const mapStateToProps = (store) => {
  return {
    alumni: store.alumData
  };
};

export default connect(mapStateToProps, null)(ViewLine);
