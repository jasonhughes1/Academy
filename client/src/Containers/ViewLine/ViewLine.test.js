import React from 'react';
import { shallow } from 'enzyme';
import { ViewLine, mapStateToProps } from './ViewLine';


describe('ViewLine Tests', () => {

  it('should be defined', () => {
    const renderedViewLine = shallow(<ViewLine alumni= {[]}/>);

    expect(renderedViewLine).toBeDefined();
  });

  it('should match the snap shot', () => {
    const renderedViewLine = shallow(<ViewLine alumni= {[]}/>);

    expect(renderedViewLine).toMatchSnapshot();
  });
});
