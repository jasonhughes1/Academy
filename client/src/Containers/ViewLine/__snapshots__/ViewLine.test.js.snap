// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewLine Tests should match the snap shot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViewLine
    alumni={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="link-container-view"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    exact={true}
                    to="/"
          >
                    Home
          </NavLink>
</div>,
        <h2
          className="view-salary-title"
>
          View salary information for cohorts from 2015 and 2016
</h2>,
        <Connect(LineGraph15)
          alumData={Array []}
/>,
        <div
          className="place-holder"
/>,
        <Connect(LineGraph16)
          alumData={Array []}
/>,
        <div
          className="place-holder"
/>,
        <Connect(LineGraphYearly)
          alumData={Array []}
/>,
        <div
          className="place-holder"
/>,
      ],
      "className": "view-line-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav"
            exact={true}
            to="/"
>
            Home
</NavLink>,
          "className": "link-container-view",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Home",
            "className": "nav",
            "exact": true,
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "View salary information for cohorts from 2015 and 2016",
          "className": "view-salary-title",
        },
        "ref": null,
        "rendered": "View salary information for cohorts from 2015 and 2016",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alumData": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "place-holder",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alumData": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "place-holder",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alumData": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "place-holder",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="link-container-view"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        exact={true}
                        to="/"
            >
                        Home
            </NavLink>
</div>,
          <h2
            className="view-salary-title"
>
            View salary information for cohorts from 2015 and 2016
</h2>,
          <Connect(LineGraph15)
            alumData={Array []}
/>,
          <div
            className="place-holder"
/>,
          <Connect(LineGraph16)
            alumData={Array []}
/>,
          <div
            className="place-holder"
/>,
          <Connect(LineGraphYearly)
            alumData={Array []}
/>,
          <div
            className="place-holder"
/>,
        ],
        "className": "view-line-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              exact={true}
              to="/"
>
              Home
</NavLink>,
            "className": "link-container-view",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Home",
              "className": "nav",
              "exact": true,
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "View salary information for cohorts from 2015 and 2016",
            "className": "view-salary-title",
          },
          "ref": null,
          "rendered": "View salary information for cohorts from 2015 and 2016",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alumData": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "place-holder",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alumData": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "place-holder",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alumData": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "place-holder",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
